src/OpenFeature.Contrib.Providers.Flagd/Resolver/InProcess/JsonEvaluator.cs

```diff
@@ -88,7 +83,57 @@ internal FlagSyncData Parse(string flagConfigurations)
                 });
             }
 
-            return JsonConvert.DeserializeObject<FlagSyncData>(transformed);
+
+            var data = JsonConvert.DeserializeObject<FlagSyncData>(transformed);
+            if (data.Metadata == null)
+            {
+                data.Metadata = new Dictionary<string, object>();
+            }
+            else
+            {
+                foreach (var key in new List<string>(data.Metadata.Keys))
```

Comment:
nit: I think it would be more typical to do this functionally with Linq (`.Select`) but that's a nit.